lib/rspec/core/world.rb

```diff
@@ -184,7 +192,9 @@ def announce_filters
         return unless example_count.zero?
 
         example_groups.clear
-        if filter_manager.empty?
+        if rspec_is_quitting
+          report_filter_message("Unable to load examples because of SystemExit.")
+        elsif filter_manager.empty?
```

Comment:
The message we want to surpress is the false "no examples found", theres already an explanation of whats happening above this, and actually if a SystemExit is encountered during a spec run there would be an example count.

```suggestion
        if filter_manager.empty? && !rspec_is_quitting
```

Reply:
> Sugesstion: `if filter_manager.empty? && !rspec_is_quitting`

I considered that, but In my testing, it produced the output: "All examples were filtered out", which seemed misleading.

> .. if a SystemExit is encountered during a spec run there would be an example count.

In the scenario that I'm testing, the `exit` occurs in a `--require` (e.g. at the top of 'spec_helper'). Would there still be a count in that case?

Reply:
Hm fair, prehaps it should be:
```suggestion     
        unless rspec_is_quitting
          if filter_manager.empty?
            report_filter_message("No examples found.")
          elsif exclusion_filter.empty? || inclusion_filter.empty?
            report_filter_message(everything_filtered_message)
          end
        end
```